m4/angry_bird.py

```
@@ -0,0 +1,241 @@
+        if len(self.birds) == 0:
+            return
+            
+        bird = self.birds[0]
+        
+        # Bug 3: 碰撞检测逻辑错误
+        for pig in self.pigs:
+            if pig.alive:
+                distance = math.sqrt((bird.x - pig.x)**2 + (bird.y - pig.y)**2)
+                if distance < bird.radius + pig.radius:
+                    pig.alive = False
+                    self.birds.remove(bird)  # 错误：直接删除可能导致索引问题
```

Ruff (0.13.1)

Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)